enhancements/installer/aws-custom-dns.md

```diff
@@ -0,0 +1,476 @@
+solution [Route53 for AWS](https://aws.amazon.com/route53/). OpenShift will
+continue to configure and manage the cloud load balancing solution [Elastic
+Load Balancer](https://aws.amazon.com/elasticloadbalancing/) for all its
+services.
+
+OpenShift will start static CoreDNS pods to provide DNS resolution for API,
+Ingress API and Ingress services that are essentail for cluster creation.
+Although not identical, this approach leverages learning from OpenShift's
+approach to providing these services for on-prem platforms.
+
+### Workflow Description
+
+1. For an OpenShift administrator to use a custom DNS solution, they would
+have to enable a config option in the install configuration. This would
+cause the Installer and the Ingress Operator to skip configuring the cloud
+DNS with entries for API, Internal API and Ingress Services. 
+
+2. The Installer and the Ingress Operator will continue to configure the cloud
+Load Balancers on the public subnets of Public Hosted Zones for `api` and
+`*.apps`. They will continue to configure cloud Load Balancers on the private
+subnets of Private Hosted Zones for `api-int` and `*.apps` for in-cluster
+access to the Ingress service.
+
+3. The Installer also creates a ConfigMap(CloudLBConfig) to store LB API
+addresses to be used later for configuration of API and API-Int DNS records
+on the in-cluster and customer and customer DNS solutions. 
+
+4. The Installer instructs the machine-config-operator(MCO) to start a static
+CoreDNS pod on the bootstrap and master nodes to provide DNS resolution for
+the API and API-Int servers. MCO uses the information in the `CloudLBConfig`
+ConfigMap to generate the CoreFile to configure the CoreDNS instance in the pod.
+Note: This implementation differs from the on-prem case by not staring the
+keepalived or haproxy containers for load balancing.
+
+5. The Installer omits the DNS Zones within `dnses.config.openshift.io/cluster`
+to influence the behavior of the Ingress Operator by making it skip the
+configuration of cloud DNS while continuing with Load Balancer configuration.
+
+6. The Ingress Controllers managed by the cluster-ingress-operator continue
+to configure the application LBs to enable in-cluster and external accesss to
+cluster services. The Ingress Controllers will continue to update the DNSRecord
+ConfigResource(CR) with the configuration required to configure DNS for these
+services.
+
+7. The MCO will be updated to watch the DNSRecord CR for the DNS configuration
+mentioned in step 6. MCO will use the runtimecfg utility to update the CoreFile
+for the CoreDNS pod to now also include DNS configuration for both internal and
+external access to the cluster services (the `*.apps` records).
+
+8. MCO will also be required to update the CloudLBConfig ConfigMap with the DNS
+configuartion for external access to the cloud services. This takes care of all
+the DNS configuartion to bring up the cluster successfully.
```

Comment:
For ingress? I think API would already be handled by Installer in step 3--no MCO intervention needed for that.

Reply:
We need to provide DNS resolution for *.apps and that is why we need this change.